app/questionnaire/rules.py

```diff
@@ -175,10 +177,8 @@ def evaluate_when_rules(when_rules, metadata, answer_store, group_instance):
     """
     for when_rule in when_rules:
         if 'id' in when_rule:
-            when_specified_group_instance = when_rule.get('group_instance')
-            if when_specified_group_instance is not None:   # 0 is a valid group instance
-                group_instance = when_specified_group_instance
-
+            if not _answer_is_in_repeating_group(when_rule['id']):
+                group_instance = 0
```

Comment:
@AndrewTorrance Is it worth just adding an initial `and` to check that group instance isn't already 0? Just reduces the amount of time that the _answer_is_in_repeating_group(...) need to be evaluated.